### ISSUE TITLE
Use variables inside <xpath> node

### ISSUE THREAD

**Issue by lukzas** [Seddryck/NBi#574]
I've tried to search repo without luck and docs mention that variables should be usable within `<xml-source>`, but no details 🙄

Is it possible to use variables defined as below
```xml
<variables>
    <variable name="OrderDateVar">
      <script language="c-sharp">
        "1999-10-20"
      </script>
    </variable>
</variables>
```
or
```xml
<instance-settling>
        <local-variable name="AddressTypeVar" type="text">
            <item>Shipping</item>
            <item>Home</item>
        </local-variable>
 </instance-settling>
```
inside `<xml-source>` node (it's kind of pseudo code as I've tried a few approaches and none have worked)
```xml
<from>//PurchaseOrder[@OrderDate="@OrderDateVar"]/Items/Item</from>
<select attribute="PurchaseOrderNumber">../..</select>
<select attribute="PartNumber">.</select>
<select>../../Address[@Type="@AddressTypeVar"]/City</select>
```

If so, could you provide a snippet on how to do that? Otherwise, could you implement such a feature? 😉

**Comment by Seddryck** [comment]
At this moment, only the `<file>` element is supporting variables, so not the `xpath` subelements.

Yes, it should be possible to support this but not exactly this. I'd go for:

```xml
<from>~//PurchaseOrder[@OrderDate="{@OrderDateVar}"]/Items/Item</from>
<select attribute="PurchaseOrderNumber">../..</select>
<select attribute="PartNumber">.</select>
<select>~../../Address[@Type="{@AddressTypeVar}"]/City</select>
```

**Comment by Seddryck** [comment]
The `from` and `select` elements are now also supporting variables. Note that the attribute `attribute` is not. This feature is available in `1.22.0-beta0076` or at https://ci.appveyor.com/project/Seddryck/nbi/builds/30116757/artifacts
